lib/matplotlib/tests/test_ft2font.py

```diff
@@ -78,25 +75,45 @@ def test_font_fallback_chinese(fig_test, fig_ref, family_name, file_name):
         fig_test.text(0.05, .85 - 0.15*j, txt, family=test_font)
 
 
+@pytest.mark.parametrize("font_list",
+                          [['DejaVu Serif', 'DejaVu Sans'],
+                           ['DejaVu Sans Mono']],
+                         ids=["two fonts", "one font"])
+def test_fallback_missing(recwarn, font_list):
+    for family_name in font_list:
+        if not fm.findfont(fm.FontProperties(family=family_name)):
+            pytest.skip(f"Font {family_name} is missing")
+
+    fig = plt.figure()
+    fig.text(.5, .5, "Hello 🙃 World!", family=font_list)
+    fig.canvas.draw()
+    assert all(isinstance(warn.message, UserWarning) for warn in recwarn)
+    e = "Glyph 128579 (\\N{{UPSIDE-DOWN FACE}}) missing from current font {}."
+    assert (sorted([(warn.message.args) for warn in recwarn]) ==
+            sorted([(e.format(font),) for font in font_list]))
+
+
 @pytest.mark.parametrize(
     "family_name, file_name",
     [
         ("WenQuanYi Zen Hei", "wqy-zenhei"),
         ("Noto Sans CJK JP", "NotoSansCJK"),
+        ("Noto Sans TC", "NotoSansTC-Regular.otf")
     ],
 )
 def test__get_fontmap(family_name, file_name):
     fp = fm.FontProperties(family=[family_name])
+
     found_file_name = Path(fm.findfont(fp)).name
     if file_name not in found_file_name:
         pytest.skip(f"Font {family_name} ({file_name}) is missing")
 
     text = "There are 几个汉字 in between!"
-    ft = fm.get_font(
-        fm.fontManager._find_fonts_by_props(
-            fm.FontProperties(family=["DejaVu Sans", family_name])
-        )
-    )
+
+    # breaking this out to debug segfault
```

Comment:
This comment is a bit disturbing to leave in.  Can this batch of changes be reverted ?